metrics/web/docs/features/public_dashboard/01_public_dashboard.md

```diff
@@ -137,3 +161,206 @@ Once we've chosen the implementation approach, let's consider the components dia
+- add the `isAuthInitialized` field of the boolean type to the `AuthNotifier`;  
+- add the `handlePublicDashboardFeatureConfigUpdates()` to the `AuthNotifier`, so we can handle the feature state updates;
+- create the [`UserProfileViewModel`](#UserProfileViewModel) class;
+- create the [`SignInAnonymouslyUseCase`](#SignInAnonymouslyUseCase) class, which calls the `signInAnonymously()` method of the `UserRepository`;
+- create the [`AuthState`](#AuthState) enum;
+- integrate the view model, the use case, and the enum described above into the `AuthNotifier`.
+
+Let's review the following class diagram, which describes the relationship between the classes in this module:
+
+![Auth Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/public_dashboard_design/metrics/web/docs/features/public_dashboard/diagrams/public_dashboard_auth_class_diagram.puml)
+
+#### Navigation module
+
+The last thing, we should have the ability to manage the navigation of the anonymous user, that is, to know which pages they can visit and which they can't. The next subsection describes changes that we should implement to integrate the public dashboard feature.
+
+##### Navigation module changes
+
+The list of changes this module requires to integrate the public dashboard feature:
+- add the field of the `AuthState` type to the `NavigationNotifier`;
+- add the `handleAuthUpdates()` method to the `NavigationNotifier`, which should handle the updating of the `AuthState` in the `AuthNotifier`;
+- create a set of the pages' `RouteName`, which the anonymous user can visit (at the moment, it is only the `dashboard` route name);
+- add the additional condition to the `_processConfiguration()` method, which should state the following, if the user auth state is logged in anonymously, and they visit the page, whose `RouteName` is not in the anonymous `RouteName` set, then redirect to the `login` page;
+
+#### Making things work
+
+After the changes in the modules above, let's move on to integrating these changes into the application.
+
+Here are the main parts that we should add to the application:
+- auto sign in the user anonymously if the feature is enabled;
+- manage normal sign in for the anonymous user;
+- change the UI of the Metrics user menu.
+
+Let's review each of the above points in the following subsections
+
+##### Auto sign in the user anonymously
+
+To auto sign in the user anonymously, we should make the following:
+1. Find the `InjectionContainer`. 
+2. Add the `handleAuthUpdates()` method of the `NavigationNotifier` to the listener of the `AuthNotifier`.
+3. Create the listener for the `FeatureConfigNotifier` and add the `handlePublicDashboardFeatureConfigUpdates()` method of the `AuthNotifier`.
+4. Add the logic of the signing the user in anonymously depending on the public dashboard feature and the auth states to the `AuthNotifier`.
```

Comment:
:readable:
I believe it would be great to describe these steps in a bit more abstract way. More precisely, we should not describe the whole algorithm here, we should provide key points of this algorithm. For example: 
- The `handleAuthUpdates()` method of the `NavigationNotifier` should be called when the `AuthNotifier.authState` changes.

After this list of key points of this algorithm, we can explain the way of implementing some of them if there is any critical functionality/tricky moment in implementing them.